docs/howtos/ext-filters.md

```diff
@@ -0,0 +1,170 @@
+ on the Ambassador API Gateway is not recommended.  Use 
+ [Authentication](../basic-auth) instead.
+
+Ambassador can authenticate incoming requests before routing them to a backing 
+service. In this tutorial, we'll configure Ambassador to use an external third 
+party authentication service.
+
+## 1. Deploy the Authentication Service
+
+Ambassador delegates the actual authentication logic to a third party authentication service. We've written a [simple authentication service](https://github.com/datawire/ambassador-auth-service) that:
+
+- listens for requests on port 3000;
+- expects all URLs to begin with `/extauth/`;
+- performs HTTP Basic Auth for all URLs starting with `/backend/get-quote/` (other URLs are always permitted);
+- accepts only user `username`, password `password`; and
+- makes sure that the `x-qotm-session` header is present, generating a new one if needed.
+
+Ambassador routes _all_ requests through the authentication service: it relies on the auth service to distinguish between requests that need authentication and those that do not. If Ambassador cannot contact the auth service, it will return a 503 for the request; as such, **it is very important to have the auth service running before configuring Ambassador to use it.**
+
+Here's the YAML we'll start with:
+
+```yaml
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: example-auth
+spec:
+  type: ClusterIP
+  selector:
+    app: example-auth
+  ports:
+  - port: 3000
+    name: http-example-auth
+    targetPort: http-api
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example-auth
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: example-auth
+  template:
+    metadata:
+      labels:
+        app: example-auth
+    spec:
+      containers:
+      - name: example-auth
+        image: docker.io/datawire/ambassador-auth-service:2.0.0
+        imagePullPolicy: Always
+        ports:
+        - name: http-api
+          containerPort: 3000
+        resources:
+          limits:
+            cpu: "0.1"
+            memory: 100Mi
+```
+
+Note that the cluster does not yet contain any Ambassador AuthService definition. This is intentional: we want the service running before we tell Ambassador about it.
+
+The YAML above is published at getambassador.io, so if you like, you can just do
+
+```shell
+kubectl apply -f https://www.getambassador.io/yaml/demo/demo-auth.yaml
+```
+
+to spin everything up. (Of course, you can also use a local file, if you prefer.)
+
+Wait for the pod to be running before continuing. The output of `kubectl get pods` should look something like
+
+```console
+$ kubectl get pods
+NAME                            READY     STATUS    RESTARTS   AGE
+example-auth-6c5855b98d-24clp   1/1       Running   0          4m
+```
+Note that the `READY` field says `1/1` which means the pod is up and running.
+
+## 2. Configure Ambassador Authentication
```

Comment:
@mattmcclure-dw You need a `FilterPolicy` as well to actually configure Ambassador to use this `Filter`.

Something like this:

```yaml
---
apiVersion: getambassador.io/v2
kind: FilterPolicy
metadata:
  name: authentication
spec:
  rules:
  - host: "*"
    path: /*
    filters:
    - authentication
```
will enable that `Filter` to run on all requests to all hosts.

Reply:
see below